### PR TITLE
imklog: rewrite documentation

### PR DESCRIPTION
**Pull request by PascalWithopf**
Module documentation was rewritten in the new style.

### CODE REVIEW

File: source/configuration/modules/imklog.rst [hunk 1]
```diff
@@ -1,34 +1,143 @@
...
+^^^^^^^
+
+.. csv-table::
+   :header: "type", "default", "mandatory", "|FmtObsoleteName| directive"
+   :widths: auto
+   :class: parameter-table
+
+   "word", "none", "no", "``$klogpath``"
+
+???
```
Comment (PascalWithopf) [inline source/configuration/modules/imklog.rst]: I found a parameter which wasn't documented until now and needs to be looked at.

**Comment by deoren** [comment]
@PascalWithopf Do you know if there are current equivalents to these [`obsolete legacy`](http://www.rsyslog.com/doc/v8-stable/configuration/conf_formats.html) configuration format directives?

- `$DebugPrintKernelSymbols`
- `$klogUseSyscallInterface`
- `$klogSymbolsTwice`

A brief config check via `rsyslogd -N2 -f rsyslog-test.conf` suggests that there are not. I'm guessing that if those [`obsolete legacy`](http://www.rsyslog.com/doc/v8-stable/configuration/conf_formats.html) configuration format directives are needed then a user will not be able to load the module using the [`advanced`](http://www.rsyslog.com/doc/v8-stable/configuration/conf_formats.html) configuration format.

**Comment by rgerhards** [comment]
> @PascalWithopf Do you know if there are current equivalents to these obsolete legacy configuration format directives?
> 
> $DebugPrintKernelSymbols
> $klogUseSyscallInterface
> $klogSymbolsTwice

These directives are gone away:
```
	CHKiRet(omsdRegCFSLineHdlr((uchar *)"klogLocalipif", 0, eCmdHdlrGoneAway,
			NULL, NULL, STD_LOADABLE_MODULE_ID));
	CHKiRet(omsdRegCFSLineHdlr((uchar *)"debugprintkernelsymbols", 0, eCmdHdlrGoneAway,
			NULL, NULL, STD_LOADABLE_MODULE_ID));
	CHKiRet(regCfSysLineHdlr2((uchar *)"klogpath", 0, eCmdHdlrGetWord,
			NULL, &cs.pszPath, STD_LOADABLE_MODULE_ID, &bLegacyCnfModGlobalsPermitted));
	CHKiRet(omsdRegCFSLineHdlr((uchar *)"klogsymbollookup", 0, eCmdHdlrGoneAway,
			NULL, NULL, STD_LOADABLE_MODULE_ID));
	CHKiRet(omsdRegCFSLineHdlr((uchar *)"klogsymbolstwice", 0, eCmdHdlrGoneAway,
			NULL, NULL, STD_LOADABLE_MODULE_ID));
	CHKiRet(omsdRegCFSLineHdlr((uchar *)"klogusesyscallinterface", 0, eCmdHdlrGoneAway,
			NULL, NULL, STD_LOADABLE_MODULE_ID));
```
They do not error out (except for saying they are gone away), but they actually do nothing.

Maybe the doc needs to be adjusted to this.

**Comment by deoren** [comment]
> @rgerhards: Maybe the doc needs to be adjusted to this.

According to https://github.com/rsyslog/rsyslog-doc/issues/453#issuecomment-353588889 (and discussion elsewhere) we need a place to list directives that no longer exist.

We could either have a section on the original page or a separate page somewhere. If a separate page, we should probably include a link to that page from any doc that originally contained the directive. If the original page, it would probably be easier for someone with an older configuration to learn that the directive is no longer supported.

@rgerhards @PascalWithopf Perhaps "unsupported directives" or `Unsupported |FmtObsoleteName| directives` as a new section?

**Comment by deoren** [comment]
refs #448

**Comment by deoren** [comment]
I found some issues with the recent changes to imrelp.rst, impstats.rst and imptcp.rst:

```
/home/deoren/dev/rsyslog/rsyslog-doc/source/configuration/modules/impstats.rst:291: WARNING: duplicate label statistic-counter, other instance in /home/deoren/dev/rsyslog/rsyslog-doc/source/configuration/modules/imrelp.rst
/home/deoren/dev/rsyslog/rsyslog-doc/source/configuration/modules/imptcp.rst:540: WARNING: duplicate label statistic-counter, other instance in /home/deoren/dev/rsyslog/rsyslog-doc/source/configuration/modules/impstats.rst
/home/deoren/dev/rsyslog/rsyslog-doc/source/configuration/modules/imrelp.rst:343: WARNING: duplicate label statistic-counter, other instance in /home/deoren/dev/rsyslog/rsyslog-doc/source/configuration/modules/imptcp.rst
```

The conflict was not immediately obvious from CI since each PR stood alone and was not rebased against master between each merge (as a result the conflict was not seen).

This will require updating all module docs using the new format to use a unique label for the stats counter section.

I'll submit a PR for those module docs and imuxsock.rst soon.

**Comment by deoren** [comment]
> @deoren: This will require updating all module docs using the new format to use a unique label for the stats counter section.

Turns out that only the module docs already merged were affected.

**Comment by PascalWithopf** [comment]
@deoren there was no label used in this one, so I think this can be merged or is there something else?

**Comment by deoren** [comment]
@PascalWithopf There was the discussion on these two posts:

- https://github.com/rsyslog/rsyslog-doc/pull/569#issuecomment-365665899
- https://github.com/rsyslog/rsyslog-doc/pull/569#issuecomment-365669354

**Comment by PascalWithopf** [comment]
@deoren I have added an Unsupported Directives section at the end.
I think it's a good idea to keep all parameters on the module page. They are easy to find there and everything concerning the module parameters is kept on one page.

**Comment by deoren** [comment]
> @PascalWithopf: @deoren I have added an Unsupported Directives section at the end.
I think it's a good idea to keep all parameters on the module page. They are easy to find there and everything concerning the module parameters is kept on one page.

We're on the same page. While the original idea (all deprecated items in an index elsewhere) made some sense at the time, in practice it seems more useful to keep the directives on the same page

**Pull request merged**